actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -21,8 +21,14 @@ def default_render(*args)
         if block_given?
           yield(*args)
         else
-          logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
-          super
+          if no_templates?
+            logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
+            super
+          else
+            format = request.format.to_sym.upcase if request.format
```

Comment:
The chances that `request.format` is falsey is pretty unlikely -- I think a bunch of other things within AP would probably break if this happens. The reason why we have to perform some cleaning functions on the `request.format` is because it's a `Mime::Type`. Changed the behavior a little to make it more understandable, and in the syntax style of `respond_to`.